src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -36,11 +42,57 @@ var Marathon = React.createClass({
       deploymentsFetchState: States.STATE_LOADING,
       fetchState: States.STATE_LOADING,
       modalClass: null,
+      route: null,
       tasksFetchState: States.STATE_LOADING
     };
   },
 
   componentDidMount: function () {
+    var router = this.props.router;
+
+    router.on("route", function (route, params) {
+      this.setState({
+        route: {
+          name: route,
+          params: params
+        }
+      });
+    }.bind(this));
+
+    router.on("route:about", function () {
+      this.modalDestroy();
+      this.setState({
+        modalClass: AboutModalComponent
+      });
+    }.bind(this));
+
+    router.on("route:apps", function (appid) {
+      if (appid) {
+        if (this.state.activeAppId !== appid) {
+          this.modalDestroy();
+        }
+
+        this.setState({
+          activeAppId: appid,
+          activeApp: this.state.collection.get("/" + appid),
```

Comment:
That's a good point!
I'm thinking it would be cleaner to have one thing only, it seems to replicate information. If you store the app, then you can get the id `app.id`. What's your opinion?